tool/format_ignores.py

```diff
@@ -0,0 +1,228 @@
+"""Format pyright ignore comments with proper sorting and spacing.
+
+Ensures that pyright ignore comments in Python files have their entries sorted
+alphabetically and formatted with proper spacing between items.
+
+Usage: uv run tool/format_ignores.py [-h] [--pattern PATTERN] [--check] [PATH]
+"""
+
+import argparse
+import errno
+import os
+import re
+import sys
+from collections.abc import Sequence
+from pathlib import Path
+
+combined_pattern = re.compile(r"(\s*#\s*(pyright|type):\s*ignore\[)([^\]]+)(\])")
+
+
+def _sort_ignore_list(ignore_text: str, /) -> str:
+    """
+    Sort the items in a pyright ignore list and ensure proper spacing.
+
+    Parameters
+    ----------
+    ignore_text : str
+        Text inside the ignore brackets
+
+    Returns
+    -------
+    str
+        Sorted and formatted ignore list
+    """
+    # Extract items from the ignore list
+    return ", ".join(sorted(set(filter(bool, map(str.strip, ignore_text.split(","))))))
+
+
+def _process_file(
+    file_path: str | os.PathLike[str],
+    /,
+    *,
+    check_only: bool = False,
+) -> bool:
+    """
+    Process a file to sort pyright ignore comments.
+
+    Parameters
+    ----------
+    file_path : str or os.PathLike[str]
+        Path to the file to process
+    check_only : bool, default=False
+        If True, don't modify the file, just check if changes are needed
+
+    Returns
+    -------
+    bool
+        True if the file was modified or would be modified (in check mode)
+    """
+    path_obj = Path(file_path)
+    content = path_obj.read_text(encoding="utf-8")
+
+    # Process both pyright and mypy ignore comments
+    def replace_match(match: re.Match[str], /) -> str:
+        prefix = match.group(1)
+        # group(2) contains either "pyright" or "type", which we don't need separately
+        ignore_list = match.group(3)
+        suffix = match.group(4)
+        sorted_ignore = _sort_ignore_list(ignore_list)
+        return f"{prefix}{sorted_ignore}{suffix}"
```

Comment:
It's good that you're avoiding code duplication here.

But my thought process behind capturing the `type|pyright` group in isolation, is for cases like `#    type:ignore[{}]`. And if I understand correctly, then with this current code, this spacing will be kept. So I thought that we might as well correct spacing issues by formatting it as `# {match.group(2)}: ignore[{sorted_ignore}]`.

Although the explicit `suffix` group tells me that I'm probably missing something here 🤔